src/chocolatey/infrastructure/services/XmlService.cs

```diff
@@ -45,75 +45,81 @@ public XmlService(IFileSystem fileSystem, IHashProvider hashProvider)
-                            {
-                                // Check if its just a malformed document.
-                                if (ex.Message.Contains("There is an error in XML document"))
-                                {
-                                    // If so, check for a backup file and try an parse that.
-                                    if (_fileSystem.file_exists(xmlFilePath + ".backup"))
-                                    {
-                                        using (var backupStream = _fileSystem.open_file_readonly(xmlFilePath + ".backup"))
-                                        using (var backupReader = new StreamReader(backupStream))
-                                        using (var backupXmlReader = XmlReader.Create(backupReader))
-                                        {
-                                            var validConfig = (XmlType)xmlSerializer.Deserialize(backupXmlReader);
-
-                                            // If there's no errors and it's valid, go ahead and replace the bad file with the backup.
-                                            if (validConfig != null)
-                                            {
-                                                _fileSystem.copy_file(xmlFilePath + ".backup", xmlFilePath, overwriteExisting: true);
-                                            }
-
-                                            return validConfig;
-                                        }
-                                    }
-                                }
-
-                                throw;
-                            }
-                            finally
-                            {
-                                foreach (var updateFile in _fileSystem.get_files(_fileSystem.get_directory_name(xmlFilePath), "*.update").or_empty_list_if_null())
-                                {
-                                    this.Log().Debug("Removing '{0}'".format_with(updateFile));
-                                    FaultTolerance.try_catch_with_logging_exception(
-                                        () => _fileSystem.delete_file(updateFile),
-                                        errorMessage: "Unable to remove update file",
-                                        logDebugInsteadOfError: true,
-                                        isSilent: true
-                                        );
-                                }
-                            }
-                        }
-                    },
-                    "Error deserializing response of type {0}".format_with(typeof(XmlType)),
-                    throwError: true);
+            return deserialize<XmlType>(xmlFilePath, 3);
+        }
 
-                }, MUTEX_TIMEOUT),
+        public XmlType deserialize<XmlType>(string xmlFilePath, int retryCount)
+        {
+            return FaultTolerance.retry(retryCount, () => GlobalMutex.enter(

```

Comment:
If 1 is number of tries, we at some point need to fix the name of our variable (tryCount vs retryCount).

Reply:
Yes, I think that makes sense.  I have taken a note to do this.